File: Client/Assets/Scripts/Libs/Scene/SceneEntityBase/EntityRenderBase.cs
using UnityEngine;

namespace SDK.Lib
{
    /**
     * @brief 基本的渲染器，所有与显示有关的接口都在这里
     */
    public class EntityRenderBase
    {
        protected SceneEntityBase m_entity;

        public EntityRenderBase(SceneEntityBase entity_)
        {
            m_entity = entity_;
        }

        virtual public void setClientDispose()
        {

        }

        virtual public bool getClientDispose()
        {
            return m_entity.getClientDispose();
        }

        virtual public GameObject gameObject()
        {
            return null;
        }

        virtual public void setGameObject(GameObject rhv)
        {

        }

        virtual public Transform transform()
        {
            return null;
        }

        virtual public void onTick(float delta)
        {
            
        }

        virtual public void show()
        {

        }

        virtual public void hide()
        {

        }

        virtual public bool IsVisible()
        {
            return true;
        }

        virtual public void dispose()
        {
            onDestroy();
        }

        virtual public void onDestroy()
        {

        }

        virtual public void setPnt(GameObject pntGO_)
        {

        }

        virtual public GameObject getPnt()
        {
            return null;
        }

        virtual public bool checkRender()
        {
            return false;
        }
    }
}

File: PlaneWar/Client/Assets/LuaFramework/Lua/MyLua/Libs/Scene/SceneEntityBase/EntityRenderBase.lua
MLoader("MyLua.Libs.Core.GlobalNS");
MLoader("MyLua.Libs.Core.Class");
MLoader("MyLua.Libs.Core.GObject");

--[[
    @brief 基本的渲染器，所有与显示有关的接口都在这里
]]

local M = GlobalNS.Class(GlobalNS.GObject);
M.clsName = "EntityRenderBase";
GlobalNS[M.clsName] = M;

function M:ctor(entity_)
    self.mEntity = entity_;
end

function M:setClientDispose()

end

function M:getClientDispose()
    return self.mEntity:getClientDispose();
end

function M:gameObject()
    return nil;
end

function M:setGameObject(rhv)

end

function M:transform()
    return nil;
end

function M:onTick(delta)
    
end

function M:show()

end

function M:hide()

end

function M:IsVisible()
    return true;
end

function M:dispose()

end

function M:setPnt(pntGO_)

end

function M:getPnt()
    return nil;
end

function M:checkRender()
    return false;
end